Kernel/Net/Intel/E1000Registers.h

```diff
@@ -0,0 +1,359 @@
+/*
+ * Copyright (c) 2024, Leon Albrecht <leon.a@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/EnumBits.h>
+#include <AK/StdLibExtras.h>
+#include <AK/Types.h>
+#include <Kernel/Library/IORegister.h>
+
+namespace Kernel::E1000 {
+
+// https://www.intel.com/content/dam/doc/manual/pci-pci-x-family-gbe-controllers-software-dev-manual.pdf
+// Table 13-2
+enum class Register {
```

Comment:
nit: I think it would be nice to have those registers sorted by category. In later datasheets you have an even clearer separation between types of registers.
i.e. https://www.mouser.com/pdfdocs/82579datasheetvol21.pdf

Reply:
Currently its in numeric order preferring the legacy aliases
In the future we may have to support the "array-like" register types, which are ordered a bit more nicely

For now I could add a few more line separations whenever we skip a register, which acheives something similar for now